src/Cms/File.php

```diff
@@ -402,10 +415,15 @@ public function mediaToken(): string
 	/**
 	 * Returns the absolute Url to the file in the public media folder
 	 * @internal
+	 *
+	 * @param string|null $thumbName Optional thumb filename
 	 */
-	public function mediaUrl(): string
+	public function mediaUrl(string|null $thumbName = null): string
 	{
-		return $this->parent()->mediaUrl() . '/' . $this->mediaHash() . '/' . $this->filename();
+		$rootUrl     = $this->parent()->mediaUrl() . '/' . $this->mediaHash();
+		$thumbName ??= $this->filename();
+
+		return $rootUrl . '/' . $thumbName;
```

Comment:
`Nitpick`: combined with suggestion from above

```suggestion
		$url        = $this->parent()->mediaUrl() . '/' . $this->mediaHash();
		$filename ??= $this->filename();

		return $url . '/' . $filename;
```